atest/testdata/keywords/type_conversion/annotations_with_typing.robot

```diff
@@ -9,15 +10,18 @@ List
     List                      [1, 2, 3.14, -42]           [1, 2, 3.14, -42]
 
 List with params
-    List with params          []                          []
-    List with params          ['foo', 'bar']              ['foo', 'bar']
-    List with params          [1, 2, 3.14, -42]           [1, 2, 3.14, -42]
+    List with ints            []                          []
+    @{numbers}=               Create list  1  ${2}  3  -42
+    List with ints            ${numbers}                  [1, 2, 3, -42]
+    List with ints            [1, ${2}, 1313, -42]        [1, 2, 1313, -42]
+    List with enums           ['foo', 'bar']              [MyEnum.foo, MyEnum.bar]
 
 Invalid list
     [Template]                Conversion Should Fail
     List                      [1, oops]                                        error=Invalid expression.
     List                      ()                                               error=Value is tuple, not list.
-    List with params          ooops                       type=list            error=Invalid expression.
+    List with ints            ooops                       type=list            error=Invalid expression.
+    List with ints            [1, ${2}, 3.14, -42]        type=list            error=Argument 'List[integer]' got value '3.14' (float) that cannot be converted to integer: Conversion would lose precision.
```

Comment:
Yes, this is quite a tricky one. Difficult to make a concise message that still has the good info. In an earlier version I had also included the index of the element for which the conversion failed. I removed that because it felt too lengthy. I ended up with this balance where the user is shown the actual failing conversion, but has to go through the list their selves to see where it is.

Reply:
Perhaps 
> Argument 'List[integer]' included value '3.14' (float) that cannot be converted to integer

Using "included" instead of "got" would make it more explicit that it was only one part of the argument that caused the problem.

Reply:
Yeah, it would be better. We already discussed this elsewhere in these comments and agreed on a small enhancement that basically changes the erro to
> Item '1' got value ...
where `1` is the index. This is wrapped into another similar error which ought to make the final message pretty good. Anyway, error reporting can certainly be enhanced once the main functionality is in.